packages/telemetry/src/otel-context-aware-slog.js

```diff
@@ -77,15 +77,31 @@ export const makeSlogSender = async options => {
       `${options.stateDir}/${DEFAULT_CONTEXT_FILE}`,
   );
 
-  const slogSender = logCreator(
-    logRecord =>
-      logger.emit({
-        ...JSON.parse(serializeSlogObj(logRecord)),
-        severityNumber: SeverityNumber.INFO,
-      }),
+  const contextualSlogProcessor = makeContextualSlogProcessor(
+    { 'chain-id': CHAIN_ID },
     persistenceUtils,
   );
 
+  /**
+   * @param {import('./context-aware-slog.js').Slog} slog
+   */
+  const slogSender = slog => {
+    const { timestamp, ...logRecord } = contextualSlogProcessor(slog);
+
+    const [secondsStr, fractionStr] = String(timestamp).split('.');
+    const seconds = parseInt(secondsStr, 10);
+    const nanoSeconds = parseInt(
+      (fractionStr || String(0)).padEnd(9, String(0)),
```

Comment:
I don't trust numbers, and want to be resilient in case the `fractionStr` is more than 9 chars already.

In general I prefer string literals where possible

```suggestion
      (fractionStr.substr(0, 9) || '').padEnd(9, '0'),
```

Reply:
According to [microtime.nowDouble](https://www.npmjs.com/package/microtime#microtimenowdouble) documentation, it returns microseconds precision so I don't think we should ever get 9 decimal places right?

Reply:
Precision of the source data doesn't mean anything for the representation as an IEEE 754 number. Those tend to do weird things often enough that I wouldn't bet on there being a max of 9 digits when stringified.

Reply:
Done